sdk/communication/AzureCommunicationChat/Source/Signaling/CommunicationSignalingClient.swift

```diff
@@ -106,13 +111,64 @@ class CommunicationSignalingClient {
 }
 
 class CommunicationSkypeTokenProvider: NSObject, TrouterSkypetokenProvider {
-    var skypeToken: String?
-    func getSkypetoken(_: Bool) -> String! {
-        return skypeToken
-    }
+    /// The cached skypeToken.
+    var token: String
+
+    /// The CommunicationTokenCredential.
+    var credential: CommunicationTokenCredential
+
+    /// Called from getSkypetoken, handle token errors here.
+    var tokenRefreshHandler: TrouterTokenRefreshHandler
+
+    /// Current number of token fetch retries.
+    var tokenRetries: Int
 
-    init(skypeToken: String? = nil) {
-        self.skypeToken = skypeToken
+    /// Max number of token retries allowed.
+    let MAX_TOKEN_RETRIES: Int = 3
```

Comment:
This is not idiomatic Swift for constants. It should just be `let maxTokenRetries`.